File: crates/rhizome-treesitter/tests/fixtures/sample.cpp
#include <string>
#include <vector>

namespace myapp {

class UserService {
public:
    UserService(const std::string& name) : name_(name) {}

    std::string getName() const {
        return name_;
    }

    void setName(const std::string& name) {
        name_ = name;
    }

private:
    std::string name_;
};

struct Config {
    std::string host;
    int port;
};

enum class Status {
    Active,
    Inactive,
    Pending
};

} // namespace myapp

template <typename T>
T maxValue(T a, T b) {
    return (a > b) ? a : b;
}

void globalFunction(int x) {
    // standalone function
}

File: crates/rhizome-treesitter/tests/fixtures/sample.java
import java.util.List;
import java.util.Map;

public class UserService {
    private String name;
    private int age;

    public UserService(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setAge(int age) {
        this.age = age;
    }
}

interface Repository {
    List<Object> findAll();
    Object findById(int id);
}

enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
}
